src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py

```diff
@@ -57,27 +70,144 @@ def execute(self):
+                            total number of RSEs
+                            total number of files found for deletion
+                            number of RSEs cleaned
+                            number of files deleted
+        """
+        totalNumRses = 0
+        totalNumFiles = 0
+        numRsesCleaned = 0
+        numFilesDeleted = 0
+
+        # Call the workflow dispatcher:
+        for rse in rseList:
+            try:
+                rse = MSUnmergedRSE(rse)
+                self.plineUnmerged.run(rse)
+                msg = "\n----------------------------------------------------------"
+                msg += "\nMSUnmergedRSE: %s"
+                msg += "\n----------------------------------------------------------"
+                self.logger.debug(msg, pformat(rse))
+                totalNumRses += 1
+                if self._checkClean(rse):
+                    numRsesCleaned += 1
+                totalNumFiles += rse['counters']['totalNumFiles']
+                numFilesDeleted += rse['counters']['numFilesDeleted']
+            except Exception as ex:
+                msg = "%s: General error from pipeline. RSE: %s. Error:  \n%s. "
+                msg += "\nWill retry again in the next cycle."
+                self.logger.exception(msg, self.plineUnmerged.name, rse['rse'], str(ex))
+                continue
+        return totalNumRses, totalNumFiles, numRsesCleaned, numFilesDeleted
+
+    def filterRseList(self, rseList):
+        """
+        Applies any additional constraints about the subset of RSEs on which
+        the current service instance is about to work
+        :param rseList: Initial list of RSEs from which the final set should be build
+        :return:        A list of RSEs
+        """
+        return rseList
+
+    def cleanFiles(self, rse):
+        """
+        The method to implement the actual deletion of files for an RSE.
+        :param rse: MSUnmergedRSE object to be cleaned
+        :return:    The MSUnmergedRSE object
+        """
+        try:
+            # for fileUnmerged in rse['files']['toDelete']:
+            #     try:
+            #         self.gfalCommand(rse['delInterface'], fileUnmerged)
+            #         rse['counters']['numFilesDeleted'] += 1
+            #         rse['files']['deletedSuccess'].append(fileUnmerged)
+            #     except Exception as ex:
+            #         rse['files']['deletedFail'].append(fileUnmerged)
+            #         msg = "Error while trying to delete file: %s for RSE: %s"
+            #         msg += "Will retry in the next cycle. Err: %s"
+            #         self.logger.debug(msg, fileUnmerged, rse['name'], str(ex))
+            rse['isClean'] = self._checkClean(rse)
+        except Exception as ex:
+            msg = "Error while cleaning RSE: %s"
+            msg += "Will retry in the next cycle. Err: %s"
+            self.logger.debug(msg, rse['name'], str(ex))
+
+        return rse
+
+    def _checkClean(self, rse):
```

Comment:
What are your plans if an RSE is marked as not cleaned? My point is, perhaps we could have a different evaluation here to avoid trying to delete 10k files when only a handful of them were missed.

Reply:
Actually I plan to do exactly nothing, since  we are going to stumble on this RSE on a next cycle only once the results of the current iteration is reflected everywhere down the chain. So the next time we'll have to deal with this same RSE we won't iterate through the already deleted files. 

It may happen that we delete all the files put for deletion for the current cycle, but in case the RSE was  too polluted and number of files have exceded the limit of deletions per RSE fora  a single cycle the ... well... this is definitely an RSE we may not consider clean and will have to iterate through it again start digesting what ever was an excess from the previous run. 

Or an equivalent scenario - we have some broken files to the level no deletion can happen even though they are present in the namespace. Then ... whell this is something that would need a manual intervention and untill that   happens the RSE shouldn't be considered clean either. And we need to know somehow about this case. I've not yet figured it out how to identify and signal it properly though. Any suggestions are welcome.

Reply:
My suggestion is to actually define a fraction where an RSE is considered clean. Like, if >75% (or 90%?) of the deletable files managed to be deleted, then we should consider that RSE clean, otherwise we will simply add extra load on our microservice and on the RSE storage.

I fear that the microservice will spend a long time trying to delete some final problematic files, and this is something that we should avoid IMO.

To be explicit, what I'm trying to avoid here is to iterate over a list of 100k files, in a given RSE, when you only need to delete 500 of them. I would say, in such cases, it would be better to wait for a new dump of the consistency check tool.

Just some thoughts. The True/False might not be enough for us.

Reply:
thanks @amaltaro 

> To be explicit, what I'm trying to avoid here is to iterate over a list of 100k files, in a given RSE, when you only need to delete 500 of them. 

As I mentioned in my previous reply, this effect should be mitigated by the `limitFilesPerRSE` configuration parameter. 

> I would say, in such cases, it would be better to wait for a new dump of the consistency check tool.

And again my point exactly - when I said:  

> we are going to stumble on this RSE on a next cycle only once the results of the current iteration is reflected everywhere down the chain

So we  are on the same page. Only that all these changes I have foreseen in the next issue, where the actual per RSE load and work would be estimated and executed, At that moment we will have a better view the situation and how to organize those limits. 

And, yes we can think of a non boolean resolution of the question what actions to be taken towards a partially cleaned RSEs.

Reply:
BTW we should make a note and link the next issue https://github.com/dmwm/WMCore/issues/10410 to this very comment so we  do not forget this.

Reply:
> As I mentioned in my previous reply, this effect should be mitigated by the limitFilesPerRSE configuration parameter.

The only way I can think of, to mitigate it, would be if we kept the list of files that we have successfully deleted persisted somewhere. Otherwise we have no idea which of the 100k files (from the example above) have been already deleted or not.
Maybe something to be considered when we attach this service to a database.

All right, let's keep discussing it in your next pull request then, once you implement the real file deletion logic.

Thanks for this follow up, Todor.